packages/adapter-api/src/values.ts

```diff
@@ -261,6 +261,7 @@ export const compareSpecialValues = (
 ): boolean | undefined => {
   if (isStaticFile(first) && isStaticFile(second)) {
     return first.isEqual(second)
+      && (options?.compareReferencesByValue !== true ? first.filepath === second.filepath : true)
```

Comment:
readability:
```suggestion
      && (options?.compareReferencesByValue !== true
        ? first.filepath === second.filepath
        : true)
```
or (I think slightly better)
```suggestion
      && (options?.compareReferencesByValue
        ? true
        : first.filepath === second.filepath)
```